hack/tools/go.mod

```diff
@@ -6,11 +6,11 @@ require (
 	github.com/aojea/sloppy-netparser v0.0.0-20210819225411-1b3bd8b3b975
 	github.com/cespare/prettybench v0.0.0-20150116022406-03b8cfe5406c
 	github.com/client9/misspell v0.3.4
-	github.com/golang/mock v1.4.4
-	github.com/golangci/golangci-lint v1.41.1
+	github.com/golang/mock v1.6.0
```

Comment:
maybe that's not required for this PR if things compile and tests pass, but that would be a good followup

Reply:
it is actually passing all tests

https://github.com/golang/mock#installation

> If you use mockgen in your CI pipeline, it may be more appropriate to fixate on a specific mockgen version. You should try to keep the library in sync with the version of mockgen used to generate your mocks.

we can probably afford to sync this in a follow-up

Reply:
agreed

Reply:
Sorry for the slow response, I am on vacation ATM.

Addressing this in https://github.com/kubernetes/kubernetes/pull/109092